app/models/cart.rb

```diff
@@ -32,20 +32,28 @@ def get_items # rubocop:disable AccessorMethodName
   end
 
   # Adds equipment model id to items hash
-  def add_item(equipment_model)
+  def add_item(equipment_model, _quantity = nil)
     return if equipment_model.nil?
     key = equipment_model.id
     items[key] = items[key] ? items[key] + 1 : 1
   end
 
   # Remove equipment model id from items hash, or decrement its count
-  def remove_item(equipment_model)
+  def remove_item(equipment_model, _quantity = nil)
     return if equipment_model.nil?
     key = equipment_model.id
     items[key] = items[key] ? items[key] - 1 : 0
     self.items = items.except(key) if items[key] <= 0
   end
 
+  def edit_cart_item(equipment_model, quantity)
+    return if equipment_model.nil?
+    return if quantity < 0
+    key = equipment_model.id
+    items[key] = items[key] ? quantity : 0
```

Comment:
I'm wondering if we need to do this; frankly, we could use the `edit_cart_item` method to add a new item to the cart, even if it wasn't there to begin with. Maybe we'd be better off with a different method name (e.g. `set_cart_item` or something like that) but I don't see why we have to restrict this to only modifying existing cart items.